frontend_lib/README.md

```diff
@@ -1,32 +1,65 @@
 # Tracim lib
 
-This project is used to centralize generic components that are used either in Tracim and in Tracim's apps.
+This library groups the components shared across the Tracim frontend and the Tracim frontend apps.
 
-### For development
-While working on Tracim and on tracim_lib, if you change tracim_lib, for the changes to be effective in Tracim you must:
-- run `$ npm run build` on tracim_lib
-- add the generated dist/tracim_lib.js to git stage
-- commit and push changes to tracim_lib
-- increase version number of tracim_lib (package.json)
-- commit push the new version
-- run `$ npm update tracim_lib` in Tracim to get the new version
+It can be built in two different ways:
 
-The debug.js file is used for all app debug files and it's based at debug.js.sample file. To adapt the template to your system you must change the variables:
- - loggedUser: object in which all parameters must be changed in relation to the current user logged on in Tracim
+- as a standalone module, that could be published on NPM and used by external apps
+- as a library to be inserted in an HTML page using the a `<script>` tag. This is what is used for Tracim.
 
-#### Alternatively you can (faster for development)
-- create a link of tracim_lib using npm: in tracim_lib `$ npm link`
-- get that link in tracim: in tracim (repo) `$ npm link tracim_lib`
-- in tracim_lib: run `$ npm run build`
+# Building as a standalone bundle
 
-now, you only have to rebuild tracim_lib and tracim will update it's tracim_lib dependency automatically
+The standalone bundle can be used in external Tracim apps as an UMD module.
+It includes all its dependencies, is self-contained, and can be imported using
+`require('tracim_frontend_lib')` or `import ... from 'tracim_frontend_lib'`.
 
-Problem: eslint will parse tracim_lib.js since it will come from another repository.
+Run:
 
-Solution: you must add `/* eslint-disable */` at the beginning and  `/* eslint-enable */` at the end of tracim_lib.js
+    yarn run build
 
-To automatize this: build tracim_lib with that command:
+This will produce two UMD modules:
+- `dist/tracim_frontend_lib.lib.js`: the library part
+- `dist/tracim_frontend_lib.style.js`: the styling part
 
-`npm run build && echo '/* eslint-disable */' | cat - dist/tracim_lib.js > temp && mv temp dist/tracim_lib.js && printf '\n/* eslint-enable */\n' >> dist/tracim_lib.js`
+The Webpack configuration used to build the standalone bundle is `webpack.config.js`.
 
-You can also create an alias for this in your .bashrc
+# Building as a browser library
```

Comment:
I'm not convinced by the name "browser library". To me, it would sound more like an addon or an extension for browser

Reply:
I'm open to suggestions :-)

Reply:
"Building as a script that will be included in a web page"?

Reply:
I tried to be more explicit, let me know if you like it